### ISSUE TITLE
a "true" tooltip on context-bookmarkpage

### ISSUE THREAD

**Issue by yfdyh000** [Quicksaver/The-Fox--Only-Better#124]
Just a little flaw.

Open the New Tab page, mouseover the bookmark button of page context menu.
![2015-10-07 22 38 31](https://cloud.githubusercontent.com/assets/1769875/10340872/aed473de-6d44-11e5-9498-550230d49897.png)

Since the 87c62ce0298a80c312ba2100949397a9b91acb11.

**Comment by Quicksaver** [comment]
Thanks for letting me know! I wonder how this happened, very curious...

**Comment by Quicksaver** [comment]
Beta version 1.3.2b2 should fix this issue if you'd like to try it out: https://github.com/Quicksaver/The-Fox--Only-Better/releases/tag/v1.3.2b2

**Comment by yfdyh000** [comment]
It looks good, although I don't understand how it works.

**Comment by Quicksaver** [comment]
Do you mean you don't know how to install it? In that link I posted click the bold link that reads "thefoxonlybetter.quicksaver.xpi" and firefox should prompt you to install it.

**Comment by yfdyh000** [comment]
No, I mean, I'm curious about the working principle of this code.

**Comment by Quicksaver** [comment]
Ah! :)

The bug was in attributes.jsm, it was a method that I wrote a long time ago and had a bug that never manifested until now. Basically, the star icon has an empty tooltip when it's disabled (which obviously Firefox knows not to show the tooltip if it's empty). That method was treating the empty string as if it didn't exist. But it knew that the tooltip was being set, and it defaults to a value of "true" when it isn't given an actual value to set:
```Javascript
toggleAttribute(element, 'tooltip', ok, 'Bookmark me'); // sets "Bookmark me" to tooltip; correct
toggleAttribute(element, 'tooltip', ok); // sets "true" to tooltip; correct
toggleAttribute(element, 'tooltip', ok, ''); // sets "true" to tooltip; incorrect!
```
 I just "taught" it to properly recognize the empty string:
```Javascript
toggleAttribute(element, 'tooltip', ok, ''); // sets "" to tooltip; correct!
```

The rest of the changes (in Watchers.jsm) aren't really about this bug, I just took the opportunity to refactor some code a little while I was looking into it. That file does exactly the same thing as before, only with about 1kb less of code.
